src/openrct2/rct2/S6Exporter.cpp

```diff
@@ -1082,97 +1040,107 @@ void S6Exporter::ExportSpriteCommonProperties(RCT12SpriteBase* dst, const Sprite
     dst->sprite_direction = src->sprite_direction;
 }
 
-void S6Exporter::ExportSpriteVehicle(RCT2SpriteVehicle* dst, const Vehicle* src)
+template<> void S6Exporter::ExportEntity(RCT12SpriteBase* dst, const Vehicle* src)
 {
+    auto* cdst = static_cast<RCT2SpriteVehicle*>(dst);
```

Comment:
The change from dst to cdst makes the diff quite a bit larger. Could we perhaps name the `RCT12SpriteBase` object `origDst` or `baseDst` or something like this, so that the `RCT2SpriteVehicle` can be renamed back to `dst`, avoiding a large diff?

Reply:
I've cleaned it up and removed the whole base and specific so it should be easier to follow now.